### ISSUE TITLE
System.FormatException: Input string was not in a correct format.

### ISSUE THREAD

**Issue by Bia10** [spectresystems/spectre.console#309]
Hello, first of all great work, this package is very addictive to use.

I got this error when accidentally trying to markup some json string.

![image](https://user-images.githubusercontent.com/2548763/111923581-3431a000-8aa0-11eb-99fb-7ec4aaf6bfa1.png)

actually this happened as accident my program accidentally attempted to parse already parsed json file.

the file text being fed into markup is this

`{"Marker":{"fileMarker":"FacWalk V101 "},"Header":{"terrain":1,"outdoor":1,"flippable":1,"width":12,"height":12,"entryCount":105},"Entrys":[{"x":10,"y":10,"walkable":1},{"x":10,"y":9,"walkable":0},{"x":10,"y":8,"walkable":0},{"x":10,"y":7,"walkable":0},{"x":10,"y":6,"walkable":1},{"x":10,"y":5,"walkable":0},{"x":10,"y":4,"walkable":0},{"x":10,"y":3,"walkable":0},{"x":9,"y":3,"walkable":0},{"x":9,"y":2,"walkable":0},{"x":8,"y":1,"walkable":0},{"x":7,"y":1,"walkable":0},{"x":7,"y":0,"walkable":0},{"x":6,"y":0,"walkable":0},{"x":6,"y":1,"walkable":0},{"x":5,"y":1,"walkable":0},{"x":5,"y":0,"walkable":1},{"x":4,"y":0,"walkable":1},{"x":4,"y":1,"walkable":0},{"x":3,"y":1,"walkable":0},{"x":2,"y":1,"walkable":1},{"x":2,"y":2,"walkable":1},{"x":3,"y":2,"walkable":0},{"x":4,"y":2,"walkable":0},{"x":5,"y":2,"walkable":0},{"x":6,"y":2,"walkable":0},{"x":7,"y":2,"walkable":0},{"x":8,"y":2,"walkable":0},{"x":8,"y":3,"walkable":0},{"x":7,"y":3,"walkable":0},{"x":6,"y":3,"walkable":0},{"x":5,"y":3,"walkable":0},{"x":4,"y":3,"walkable":0},{"x":3,"y":3,"walkable":0},{"x":1,"y":3,"walkable":0},{"x":1,"y":4,"walkable":1},{"x":2,"y":4,"walkable":1},{"x":2,"y":3,"walkable":1},{"x":3,"y":4,"walkable":0},{"x":4,"y":4,"walkable":0},{"x":5,"y":4,"walkable":0},{"x":6,"y":4,"walkable":0},{"x":7,"y":4,"walkable":0},{"x":8,"y":4,"walkable":0},{"x":9,"y":4,"walkable":0},{"x":9,"y":5,"walkable":0},{"x":8,"y":5,"walkable":0},{"x":7,"y":5,"walkable":0},{"x":6,"y":5,"walkable":0},{"x":5,"y":5,"walkable":0},{"x":4,"y":5,"walkable":0},{"x":3,"y":5,"walkable":0},{"x":2,"y":5,"walkable":1},{"x":2,"y":6,"walkable":0},{"x":3,"y":6,"walkable":0},{"x":4,"y":6,"walkable":0},{"x":5,"y":6,"walkable":0},{"x":6,"y":6,"walkable":0},{"x":7,"y":6,"walkable":0},{"x":8,"y":6,"walkable":0},{"x":9,"y":7,"walkable":0},{"x":9,"y":6,"walkable":0},{"x":8,"y":7,"walkable":0},{"x":7,"y":7,"walkable":0},{"x":6,"y":7,"walkable":0},{"x":5,"y":7,"walkable":0},{"x":4,"y":7,"walkable":0},{"x":3,"y":7,"walkable":0},{"x":2,"y":7,"walkable":0},{"x":1,"y":7,"walkable":0},{"x":0,"y":7,"walkable":1},{"x":0,"y":8,"walkable":1},{"x":1,"y":8,"walkable":0},{"x":2,"y":8,"walkable":0},{"x":3,"y":8,"walkable":0},{"x":4,"y":8,"walkable":0},{"x":5,"y":8,"walkable":0},{"x":6,"y":8,"walkable":0},{"x":7,"y":8,"walkable":0},{"x":8,"y":8,"walkable":0},{"x":9,"y":8,"walkable":0},{"x":9,"y":9,"walkable":0},{"x":8,"y":9,"walkable":0},{"x":7,"y":9,"walkable":0},{"x":6,"y":9,"walkable":0},{"x":5,"y":9,"walkable":0},{"x":4,"y":9,"walkable":0},{"x":3,"y":9,"walkable":0},{"x":2,"y":9,"walkable":1},{"x":1,"y":9,"walkable":1},{"x":2,"y":10,"walkable":1},{"x":3,"y":10,"walkable":0},{"x":4,"y":10,"walkable":0},{"x":5,"y":10,"walkable":0},{"x":6,"y":10,"walkable":0},{"x":7,"y":10,"walkable":0},{"x":8,"y":10,"walkable":0},{"x":9,"y":10,"walkable":0},{"x":9,"y":11,"walkable":0},{"x":8,"y":11,"walkable":0},{"x":7,"y":11,"walkable":0},{"x":6,"y":11,"walkable":0},{"x":4,"y":11,"walkable":1},{"x":11,"y":8,"walkable":1},{"x":11,"y":5,"walkable":1}]}
`

the code supposed to log it is this:

```c#
public static void Log(string message, string severity)
{
    var year = DateTime.Now.Year.ToString("0000");
    var month = DateTime.Now.Month.ToString("00");
    var day = DateTime.Now.Day.ToString("00");
    var hour = DateTime.Now.Hour.ToString("00");
    var minute = DateTime.Now.Minute.ToString("00");
    var second = DateTime.Now.Second.ToString("00");
    var milliseconds = DateTime.Now.Millisecond.ToString("000");
    var timeStampFull = year + "/" + month + "/" + day + "-" + hour + ":" + minute + ":" + second + ":" + milliseconds;

    try
    {
        switch (severity)
        {

            case "success":
                AnsiConsole.MarkupLine($"[green]{timeStampFull}-{severity.ToUpper()}:[/] {message} [green]...[/]");
                break;
            case "debug":
                AnsiConsole.MarkupLine($"[grey]{timeStampFull}-{severity.ToUpper()}:[/] {message} [grey]...[/]");
                break;
            case "info":
                AnsiConsole.MarkupLine($"[white]{timeStampFull}-{severity.ToUpper()}:[/] {message} [white]...[/]");
                break;
            case "warn":
                AnsiConsole.MarkupLine($"[darkorange]{timeStampFull}-{severity.ToUpper()}:[/] {message} [darkorange]...[/]");
                break;
            case "error":
                AnsiConsole.MarkupLine($"[red1]{timeStampFull}-{severity.ToUpper()}:[/] {message} [red1]...[/]");
                break;
            case "critical":
                AnsiConsole.MarkupLine($"[red3_1]{timeStampFull}-{severity.ToUpper()}:[/] {message} [red3_1]...[/]");
                break;

            default:
                var ex = new Exception("Unsupported log severity!");
                AnsiConsole.WriteException(ex);
                break;
        }
    }
    catch (Exception e)
    {
        AnsiConsole.WriteException(e);
        throw;
    }
}
```

the place where its called:
```c#
        public FacWalk Read()
        {
            var obj = new FacWalk
            {
                Marker = MarshalUtils.ByteArrayToStructure<FacWalkMarker>(_reader)
            };
            AnsiConsoleExtensions.Log($"Parsed file marker: {obj.Marker.fileMarker}", "success");
...
```

So TLDR:
The json file gets marshaled according to FacWalkMarker structure is then fed into 
```c#
AnsiConsoleExtensions.Log($"Parsed file marker: {obj.Marker.fileMarker}", "success");
```
which calls
```c#
 AnsiConsole.MarkupLine($"[green]{timeStampFull}-{severity.ToUpper()}:[/] {message} [green]...[/]");
```
and exception is thrown.

The code works just fine on the original game data
![image](https://user-images.githubusercontent.com/2548763/111923954-28df7400-8aa2-11eb-9f0b-7be751e969af.png)

So I am rly thrilled to know what's wrong or why it cannot handle the mangled json string data.
Thanx

**Comment by phil-scott-78** [comment]
Thanks for the kind words about the library, @Bia10! Double check the version is the latest because this sounds like a bug that was just recently fixed. 

If you are still seeing maybe it wasn't fully fixed 😯 but with your reproduction I could take a look early this week if the new version doesn't fix it

**Comment by Bia10** [comment]
well I am using the latest release:

<PackageReference Include="Spectre.Console" Version="0.38.0" />

**Comment by phil-scott-78** [comment]
Ok, I see the issue. Looks like my hunch about not fixing the bug originally looks to be true. 

So, the original issue was this: we always were calling `string.Format` even when there were no parameters to format. Since you are passing in json it is seeing the brackets and throwing the exception. I missed the helpers on the static `AnsiConsole` instance to include a few overrides to skip that `string.Format`. 

I'll work on a fix.

Thanks for the thorough example and documentation, very much appreciated!
